torchgeo/datasets/utils.py

```diff
@@ -421,6 +424,54 @@ def _dict_list_to_list_dict(
     return uncollated
 
 
+def pad_across_batches(
+    batch: list[dict[str, Tensor]],
+    padding_value: float = 0.0,
+    padding_length: int | None = None,
```

Comment:
Thought about this more and it seems like `None` is just a wrapper around `pad_sequence`. Let's remove support for `None` and people can instead use `pad_sequence` directly if that's the behavior they want.